Agents/AssetManagerAgent/AssetManagerAgent/src/main/java/uk/ac/cam/cares/jps/agent/assetmanager/AssetManagerAgent.java

```diff
@@ -0,0 +1,929 @@
+                String IRI = assetData.getString("IRI");
+                jsonMessage = contactPrintServer(args, IRI);
+            }
+            else if(urlPath.contains("addmaintenance")){
+                jsonMessage = addMantainanceData(args, assetData);
+            }
+            else if(urlPath.contains("updatetime")){
+                jsonMessage = updateMaintenanceTime();
+            }
+            else if(urlPath.contains("deletemaintenance")){
+                jsonMessage = deleteMaintenanceSchedule(args, assetData.getString("ID"));
+            }
+            else if(urlPath.contains("delete")){
+                jsonMessage = deleteAsset(args, assetData.getString("ID"));
+            }
+
+            jsonMessage.accumulate("Result", "Command Success");
+            requestParams = jsonMessage;
+        }
+        else {
+            jsonMessage.put("Result", "Request parameters are not defined correctly.");
+            requestParams = jsonMessage;
+        }
+        return requestParams;
+    }
+
+    //Read properties file
+    private void readPropFile (String propFile) throws IOException{
+        try (InputStream input = new FileInputStream(propFile)) {
+            // Load properties file from specified path
+            Properties prop = new Properties();
+            prop.load(input);
+
+            try {
+                // Read the mappings folder from the properties file
+                ENDPOINT_KG_ASSET = prop.getProperty("endpoint.kg.asset");
+                ENDPOINT_KG_OFFICE= prop.getProperty("endpoint.kg.office");
+                ENDPOINT_KG_LAB= prop.getProperty("endpoint.kg.lab");
+                ENDPOINT_KG_PURCHASEDOC  = prop.getProperty("endpoint.kg.purchasedocs");
+                ENDPOINT_KG_BMS = prop.getProperty("endpoint.kg.bms");
+                KG_USERNAME = prop.getProperty("auth.kg.user");
+                KG_PASSWORD = prop.getProperty("auth.kg.pass");
+                ENDPOINT_PRINTER = prop.getProperty("endpoint.printer");
+                TARGET_QR_SIZE = Double.parseDouble(prop.getProperty("target_qr_size"));
+                URL_MANUAL = prop.getProperty("url.manual");
+                URL_DOCUPLOAD = prop.getProperty("url.docupload");
+            }
+            catch (Exception e) {
+                throw new IOException ("The endpoint keys cannot be retrieved from the properties file: ", e);
+            }
+            
+
+        }
+        catch (Exception e) {
+            throw new JPSRuntimeException("Failed to read properties file: ", e);
+        }
+    }
+
+    public boolean validateInput(JSONObject requestParams, String pathURL) throws BadRequestException {
+        boolean validate = true;
+        
+        if (System.getenv(KEY_AGENTPROPERTIES) == null) {
+            return false; 
+        }
+
+        if (pathURL.contains("getuidata")){return true;} //Needed no body, do no validation
+        
+
+        validate = requestParams.has("assetData");
```

Comment:
the instantiate setData case will not pass through this check and fail even before the agent checks which path is called